src/sync/mod.rs

```diff
@@ -38,10 +37,60 @@ pub async fn begin_sync(
     client_id: String,
     sync_id: String,
 ) -> Result<BeginSyncResponse, BeginSyncError> {
-    use BeginSyncError::*;
+    let (base_snapshot, batch_push_info) = push(
+        &sync_id,
+        store,
+        lc.clone(),
+        client_id.clone(),
+        pusher,
+        PushRequest {
+            batch_push_url: begin_sync_req.batch_push_url,
+            data_layer_auth: begin_sync_req.data_layer_auth.clone(),
+        },
+    )
+    .await
+    .map_err(BeginSyncError::PushError)?;
+
+    let BeginPullResponse {
+        client_view_info,
+        sync_head,
+        sync_id,
+    } = begin_pull(
+        base_snapshot,
+        client_id,
+        BeginPullRequest {
+            client_view_url: begin_sync_req.client_view_url,
+            data_layer_auth: begin_sync_req.data_layer_auth,
+            diff_server_url: begin_sync_req.diff_server_url,
+            diff_server_auth: begin_sync_req.diff_server_auth,
+        },
+        puller,
+        sync_id,
+        store,
+        lc,
+    )
+    .await
+    .map_err(BeginSyncError::BeginPullError)?;
+
+    Ok(BeginSyncResponse {
+        sync_info: SyncInfo {
+            sync_id,
+            batch_push_info,
+            client_view_info: Some(client_view_info),
+        },
+        sync_head,
+    })
+}
 
-    let mut begin_sync_resp = BeginSyncResponse::default();
-    begin_sync_resp.sync_info.sync_id = sync_id.clone();
+pub async fn push(
+    sync_id: &str,
+    store: &dag::Store,
+    lc: LogContext,
+    client_id: String,
+    pusher: &dyn push::Pusher,
+    req: PushRequest,
+) -> Result<(Commit, Option<BatchPushInfo>), PushError> {
+    use PushError::*;
 
     // Push: find pending commits between the base snapshot and the main head
```

Comment:
Can probably remove the "push" in this comment. We're inside a function named push :)

Reply:
Moved to doc comment